types/index.d.ts

```diff
@@ -146,18 +158,16 @@ export function renderHook<
 >(
   render: (initialProps: Props) => Result,
   options?: RenderHookOptions<Props, Q, Container, BaseElement>,
-): RenderHookResult<Result, Props>
+): Promise<RenderHookResult<Result, Props>>
 
 /**
  * Unmounts React trees that were mounted with render.
  */
-export function cleanup(): void
+export function cleanup(): Promise<void>
 
 /**
- * Simply calls ReactDOMTestUtils.act(cb)
- * If that's not available (older version of react) then it
- * simply calls the given callback immediately
+ * `act` for the DOM renderer
  */
-export const act: typeof reactAct extends undefined
-  ? (callback: () => void) => void
-  : typeof reactAct
+export function act<T>(scope: () => T): Promise<T>
+
+export * from '@testing-library/dom'
```

Comment:
Add comment here and to `src/pure.js` explaining why export order is important. 

iirc we had a discussion around this with Rollup people and this was actually specified behavior: first export wins not last like you would expect if it would work like object spreading i.e. `export * from ...; export { fireEvent }` is not equivalent to `module.exports = { ...dtlExports, fireEvent }`